### PR TITLE
i18n(pt-BR): Add `endpoints.md` translation

### PR DESCRIPTION
**Pull request by peterfritz**
#### What kind of changes does this PR include?
- Translated content

#### Description
Translates [Endpoints](https://docs.astro.build/en/core-concepts/endpoints/) to pt-BR.

I am taking part in Hacktoberfest so if everything is ok with the PR, I would appreciate it beign labeled with `hacktoberfest-accepted`.

By the way, Astro is amazing. I could not thank you enough for maintaining and constantly improving it ❤️

### CODE REVIEW

**Review by wtchnm — CHANGES_REQUESTED** [review]
Hey @peterfritz, I found three typos. Great job!

File: src/pages/pt-br/core-concepts/endpoints.md [hunk 1]
```diff
@@ -0,0 +1,292 @@
+---
+layout: ~/layouts/MainLayout.astro
+title: Endpoints
+description: Aprenda a criar endpoints que podem processar todo tipo de dados
+i18nReady: true
+---
+
+O Astro permite que você crie endpoints customizados para servir e processar todo tipo de dados. Isso pode ser usado para gerar imagens, gerar e expor um arquivo RSS ou usar como rotas de API para contruir uma API completa para o seu site.
```
Comment (peterfritz) [inline src/pages/pt-br/core-concepts/endpoints.md:8]:
```suggestion
O Astro permite que você crie endpoints customizados para servir e processar todo tipo de dados. Isso pode ser usado para gerar imagens, gerar e expor um arquivo RSS ou usar como rotas de API para construir uma API completa para o seu site.
```

File: src/pages/pt-br/core-concepts/endpoints.md [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+    headers: {
+      "Content-Type": "application/json"
+    }
+  });
+}
+```
+
+Esse código será executado para cada requisição que corresponda à rota. Por exemplo, se navegarmos para `/rick-astley.json`, `params.id` será `rick-astley`.
+
+Assim, se `rick-astley` existir no banco de dados, o endpoint irá criar um objeto `Response` para responder com seu respectivo conteúdo e retornar um código de resposta de sucesso. caso contrário, ele usará o objeto `Response` para responder com um erro `404`.
```
Comment (peterfritz) [inline src/pages/pt-br/core-concepts/endpoints.md:140]:
```suggestion
Assim, se `rick-astley` existir no banco de dados, o endpoint irá criar um objeto `Response` para responder com seu respectivo conteúdo e retornar um código de resposta de sucesso. Caso contrário, ele usará o objeto `Response` para responder com um erro `404`.
```

File: src/pages/pt-br/core-concepts/endpoints.md [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+      statusText: 'Não encontrado'
+    });
+  }
+
+  return Response.redirect(link, 307);
+}
+```
+
+### Exemplo: Verificando um desafio captcha
+Endpoints dinâmicos podem ser usados como uma API REST para desenpenhar funções como autenticação, acesso ao banco de dados e validações sem expor dados sensíveis ao usuário.
```
Comment (peterfritz) [inline src/pages/pt-br/core-concepts/endpoints.md:230]:
```suggestion
Endpoints dinâmicos podem ser usados como uma API REST para desempenhar funções como autenticação, acesso ao banco de dados e validações sem expor dados sensíveis ao usuário.
```

**Comment by peterfritz** [comment]
Hi @wtchnm, yesterday I deleted the head repository, and now after spending 1 hour trying to apply the suggested changes to this PR or recover the fork, I still wasn't able to.
I'm sorry, I didn't think I was this bad at typing and Git. Should I open another PR with the commit b4adfdb and a fix for the typos?

**Comment by wtchnm** [comment]
No problem @peterfritz. I think that opening another PR will do.

**Comment by peterfritz** [comment]
Closed in favor of #1678

**Pull request closed without merging**